src/leap/mail/imap/account.py

```diff
@@ -46,107 +46,56 @@ def _debugProfiling(result, cmdname, start):
 
-        # XXX SHOULD assert too that the name matches the user/uuid with which
-        # soledad has been initialized.
+        :param account: The name of the account (user id).
+        :type account: str
 
-        # XXX ??? why is this parsing mailbox name??? it's account...
-        # userid? homogenize.
-        self._account_name = self._parse_mailbox_name(account_name)
-        self._soledad = soledad
-        self._memstore = memstore
+        :param store: a Soledad instance.
+        :type store: Soledad
 
-        self.__mailboxes = set([])
-
-        self.initialize_db()
-
-        # every user should have the right to an inbox folder
-        # at least, so let's make one!
-        self._load_mailboxes()
-
-        if not self.mailboxes:
-            self.addMailbox(self.INBOX_NAME)
-
-    def _get_empty_mailbox(self):
+        :param d: a deferred that will be fired with this IMAPAccount instance
+                  when the account is ready to be used.
+        :type d: defer.Deferred
         """
-        Returns an empty mailbox.
+        leap_assert(store, "Need a store instance to initialize")
+        leap_assert_type(store, Soledad)
 
-        :rtype: dict
-        """
-        return copy.deepcopy(self.EMPTY_MBOX)
+        # TODO assert too that the name matches the user/uuid with which
+        # soledad has been initialized. Although afaik soledad doesn't know
+        # about user_id, only the client backend.
 
-    def _get_mailbox_by_name(self, name):
-        """
-        Return an mbox document by name.
+        self.user_id = user_id
+        self.account = Account(store, ready_cb=lambda: d.callback(self))
 
-        :param name: the name of the mailbox
-        :type name: str
-
-        :rtype: SoledadDocument
-        """
-        # XXX use soledadstore instead ...;
-        doc = self._soledad.get_from_index(
-            self.TYPE_MBOX_IDX, self.MBOX_KEY,
-            self._parse_mailbox_name(name))
-        return doc[0] if doc else None
-
-    @property
-    def mailboxes(self):
-        """
-        A list of the current mailboxes for this account.
-        :rtype: set
-        """
-        return sorted(self.__mailboxes)
+    def _return_mailbox_from_collection(self, collection, readwrite=1):
+        if collection is None:
+            return None
+        mbox = IMAPMailbox(collection, rw=readwrite)
+        return mbox
 
-    def _load_mailboxes(self):
-        self.__mailboxes.update(
-            [doc.content[self.MBOX_KEY]
-             for doc in self._soledad.get_from_index(
-                 self.TYPE_IDX, self.MBOX_KEY)])
-
-    @property
-    def subscriptions(self):
-        """
-        A list of the current subscriptions for this account.
-        """
-        return [doc.content[self.MBOX_KEY]
-                for doc in self._soledad.get_from_index(
-                    self.TYPE_SUBS_IDX, self.MBOX_KEY, '1')]
+    def callWhenReady(self, cb, *args, **kw):
```

Comment:
This might need some documentation. I find a bit of confusing that cb's first argument is something undetermined and not `*args`. I would even prefer to change the implementation so the only params of cb are `*args` and `**kw`.